ui-admin/src/study/kits/KitEnrolleeSelection.tsx

```diff
@@ -0,0 +1,176 @@
+  const [isLoading, setIsLoading] = useState(true)
+  const [enrollees, setEnrollees] = useState<Enrollee[]>([])
+  const [rowSelection, setRowSelection] = useState<Record<string, boolean>>({})
+  const [columnVisibility, setColumnVisibility] = useState<VisibilityState>({})
+  const [columnFilters, setColumnFilters] = useState<ColumnFiltersState>([])
+  const [showRequestKitModal, setShowRequestKitModal] = useState(false)
+
+  const loadEnrollees = async () => {
+    setIsLoading(true)
+    const enrollees = await Api.fetchEnrollesForKitManagement(
+      portal.shortcode, study.shortcode, currentEnv.environmentName)
+    setEnrollees(enrollees)
+    setIsLoading(false)
+  }
+
+  useEffect(() => {
+    loadEnrollees()
+  }, [])
+
+  const onSubmit = async (kitType: string) => {
+    const enrolleesSelected = Object.keys(rowSelection)
+      .filter(key => rowSelection[key])
+      .map(key => enrollees[parseInt(key)].shortcode)
+
+    // This iteration should be happening server-side: JN-460
+    for (const shortcode of enrolleesSelected) {
+      await Api.createKitRequest(
+        portal.shortcode, study.shortcode, currentEnv.environmentName, shortcode, kitType)
+    }
+
+    setShowRequestKitModal(false)
+    loadEnrollees()
+  }
+
+  const numSelected = Object.keys(rowSelection).length
+  const enableActionButtons = numSelected > 0
+
+  const hasEnrolleeCompletedSurvey = (enrollee: Enrollee, survey: Survey): boolean => {
+    const task = enrollee.participantTasks.find(task => task.targetStableId === survey.stableId)
+    return task?.status === 'COMPLETE'
+  }
+
+  const studyColumns: ColumnDef<Enrollee, string | boolean>[] = currentEnv.configuredSurveys.length === 0 ? [] :
+    currentEnv.configuredSurveys
+      .map(configuredSurvey => {
+        const survey = configuredSurvey.survey
+        return {
+          id: survey.stableId,
+          header: survey.name,
+          accessorFn: data => hasEnrolleeCompletedSurvey(data, survey),
+          meta: {
+            columnType: 'boolean',
+            filterOptions: [
+              { value: true, label: 'Completed' },
+              { value: false, label: 'Not Completed' }
+            ]
+          },
+          filterFn: 'equals',
+          cell: data => hasEnrolleeCompletedSurvey(data.row.original, survey)
+            ? <FontAwesomeIcon icon={faCheck}/>
+            : ''
+        }
+      })
+
+  const columns: ColumnDef<Enrollee, string | boolean>[] = [{
+    id: 'select',
+    header: ({ table }) => <IndeterminateCheckbox
+      checked={table.getIsAllRowsSelected()} indeterminate={table.getIsSomeRowsSelected()}
+      onChange={table.getToggleAllRowsSelectedHandler()}/>,
+    cell: ({ row }) => (
+      <div className="px-1">
+        <IndeterminateCheckbox
+          checked={row.getIsSelected()} indeterminate={row.getIsSomeSelected()}
+          onChange={row.getToggleSelectedHandler()} disabled={!row.getCanSelect()}/>
+      </div>
+    )
+  }, {
```

Comment:
might not be in the scope of this PR, but this table could really use to be sortable by enrollee creation date